### PR TITLE
Save only on > 0 rules and raise exception for empty models

### PR DESCRIPTION
**Pull request by m09**
I'm not really familiar with all the latest developments in the report analyzers so at this point I only raised exceptions when trying to access inexistant rules. There are probably better specific tweaks that could be applied.

### CODE REVIEW

File: lookout/style/format/analyzer.py [hunk 1]
```diff
@@ -223,7 +223,10 @@ def train(cls, ptr: ReferencePointer, config: Mapping[str, Any], data_service: D
                 "\n\t".join("%-55s %.5E" % (fe.feature_names[i], importances[i])
                             for i in numpy.argsort(-importances)[:25] if importances[i] > 1e-5))
             # throw away imprecise classes
-            model[language] = trainable_rules.rules
+            if trainable_rules.rules.rules:
+                model[language] = trainable_rules.rules
+            else:
+                cls._log.warning("model for %s had 0 rule. Skipping" % language)
```
Comment (zurk) [inline lookout/style/format/analyzer.py]:
```suggestion
                cls._log.warning("Model have 0 rule for %s. Skipping." % language)
```
?

**Comment by m09** [comment]
I don't mind which exception to use. `NotFittedError` is fine in the sense that the model is not fitted for this language, the others are also ok.

File: lookout/style/format/analyzer.py (hunk 1 above)
Comment (m09) [reply lookout/style/format/analyzer.py]: we have really inconsistent style so I think it's really arbitrary to pick one or the other (the 2 styles seem to be 50/50 in analyzer.py). We could do a pass to harmonize at some point though

Comment (vmarkovtsev) [inline lookout/style/format/analyzer.py]:
```suggestion
                cls._log.warning("model for %s has 0 rules. Skipping" % language)
```

File: lookout/style/format/quality_report.py [hunk 2]
```diff
@@ -107,6 +108,8 @@ def generate_model_report(model: FormatModel,
     languages = languages if isinstance(languages, Iterable) else [languages]
     template = _load_jinja2_template("templates/model_report.md.jinja2")
     for language in languages:
+        if language not in model:
```
Comment (vmarkovtsev) [inline lookout/style/format/quality_report.py]: Not related but I have just noticed a typo upwards:
```
:param languages: **Lnguages** for which report
```

**Pull request merged**